polkadot/node/core/pvf/src/error.rs

```diff
@@ -39,6 +39,11 @@ pub enum ValidationError {
 	/// Preparation or execution issue caused by an internal condition. Should not vote against.
 	#[error("candidate validation: internal: {0}")]
 	Internal(#[from] InternalValidationError),
+	/// The execution deadline of allowed_ancestry_len + 1 has been reached. Jobs like backing have
+	/// a limited time to execute. Once the deadline is reached, the current candidate cannot be
+	/// backed, regardless of its validity.
+	#[error("candidate validation: execution deadline has been reached.")]
+	ExecutionDeadline,
```

Comment:
Nice PR! Would this have worked as an `InternalValidationError`? If there is back pressure, it's technically not a problem with the PVF itself, so you could have nodes abstain from voting instead of voting against. The only reason I'd be hesitant with that, is that if one node is experiencing back pressure, it's possible that the rest of the network is as well. If too many nodes abstain then IIRC it could cause a finality stall.

Reply:
OMG, that's one epic comeback!

Reply:
@mrcnski have we met before?
We have backing back pressure that should free workers to do approval job, so the effect is supposed to be opposite.